kolibri/plugins/learn/assets/src/views/CustomContentRenderer.vue

```diff
@@ -0,0 +1,219 @@
+<template>
+
+  <CoreFullscreen
+    ref="html5Renderer"
+    class="html5-renderer"
+    :style="{ height: contentRendererHeight, width: iframeWidth }"
+  >
+    <div class="iframe-container" :style="containerStyle">
+      <iframe
+        ref="iframe"
+        class="iframe"
+        sandbox="allow-scripts allow-same-origin"
+        :style="{ backgroundColor: $themePalette.grey.v_100 }"
+        frameBorder="0"
+        :src="rooturl"
+      >
+      </iframe>
+    </div>
+  </CoreFullscreen>
+
+</template>
+
+
+<script>
+
+  import pick from 'lodash/pick';
+  import urls from 'kolibri.urls';
+  import CoreFullscreen from 'kolibri.coreVue.components.CoreFullscreen';
+  import Hashi from 'hashi';
+  import { now } from 'kolibri.utils.serverClock';
+  import { ContentNodeResource } from 'kolibri.resources';
+  import router from 'kolibri.coreVue.router';
+  import { events } from 'hashi/src/hashiBase';
+
+  const defaultContentHeight = '500px';
+  const frameTopbarHeight = '37px';
+  const pxStringAdd = (x, y) => parseInt(x, 10) + parseInt(y, 10) + 'px';
+
+  export default {
+    name: 'CustomContentRenderer',
+    components: {
+      CoreFullscreen,
+    },
+    props: {
+      topic: {
+        type: Object,
+        required: true,
+      },
+    },
+    // data() {
+    //   return {
+    //     open: false,
+    //   };
+    // },
+    computed: {
+      context() {
+        const fetchedEncodedContext = this.$route.query;
+        return JSON.stringify(decodeURI(fetchedEncodedContext));
+      },
+      rooturl() {
+        return urls.hashi();
+      },
+      iframeHeight() {
+        return defaultContentHeight;
+      },
+      iframeWidth() {
+        return 'auto';
+      },
+      contentRendererHeight() {
+        return pxStringAdd(this.iframeHeight, frameTopbarHeight);
+      },
+      containerStyle() {
+        return { height: this.iframeHeight };
+      },
+    },
+    mounted() {
+      this.hashi = new Hashi({ iframe: this.$refs.iframe, now });
+      const zipFile = this.topic.files.find(f => f.extension === 'zip');
+      this.hashi.initialize({}, {}, zipFile.storage_url, zipFile.checksum);
+      this.hashi.on('collectionrequested', message => {
+        this.fetchContentCollection(message);
+      });
+      this.hashi.on('modelrequested', message => {
+        this.fetchContentModel(message);
```

Comment:
To avoid having to do `let self = this` inside the functions. (Helps out with the unused vue properties rule).

```suggestion
        this.fetchContentModel(message).bind(this);
```